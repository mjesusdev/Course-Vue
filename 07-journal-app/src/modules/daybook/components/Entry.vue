<template>
    <div 
        class="entry-container mb-3 pointer p-2"
        @click="$router.push({ name: 'entry', params: { id: 10 }})"
    >
        <!-- Title -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">18</span>
            <span class="mx-1 fs-5">Octubre</span>
            <span class="mx-2 fw-light">2021, Lunes</span>
        </div>

        <div class="entry-description">
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. 
            Velit vel natus, nihil illo sapiente accusantium ducimus commodi dolor blanditiis aliquid quis ipsam tempora quibusdam, 
            suscipit adipisci necessitatibus nulla quod. Debitis.
        </div>
    </div>
</template>

<style lang="scss" scoped>
    .entry-container {
        border-bottom: 1px solid #2c3e50;
        transition: 0.2s all ease-in;

        &:hover {
            background-color: lighten($color: grey, $amount: 45);
            transition: 0.2s all ease-in;
        }

        .entry-container {
            font-size: 12px;
        }
    }
</style>