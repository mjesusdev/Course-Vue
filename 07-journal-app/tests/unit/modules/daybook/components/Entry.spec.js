import { shallowMount } from '@vue/test-utils'

import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Tests in Entry Component', () => {

    // mockRouter
    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount( Entry, { 
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('should do match with snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('should redirect when clicks in the entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith(
            {
                name: 'entry',
                params: {
                    id: journalState.entries[0].id
                }
            }
        )
    })

    test('tests in properties computed', () => {
        expect( wrapper.vm.day ).toBe(28)
        expect( wrapper.vm.month ).toBe('Octubre')
        expect( wrapper.vm.yearDay ).toBe('2021, Jueves')
    })
})