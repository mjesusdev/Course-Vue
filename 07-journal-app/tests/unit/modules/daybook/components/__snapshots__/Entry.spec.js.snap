// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Entry Component should do match with snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Title -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">28</span><span class="mx-1 fs-5">Octubre</span><span class="mx-2 fw-light">2021, Jueves</span></div>
  <div class="entry-description">Hola Mundo, en mi DayBook jeje ðŸ¤—</div>
</div>
`;
